packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1424,6 +1424,39 @@ class Transform extends SingleChildRenderObjectWidget {
         assert(scale == null || (scaleX == null && scaleY == null), "If 'scale' is non-null then 'scaleX' and 'scaleY' must be left null"),
         transform = Matrix4.diagonal3Values(scale ?? scaleX ?? 1.0, scale ?? scaleY ?? 1.0, 1.0);
 
+  /// Creates a widget that flips/mirrors its child along the 2D plane.
+  ///
+  /// If `flipX` is true, the child widget will be flipped horizontally. Defaults to false.
+  ///
+  /// If `flipY` is true, the child widget will be flipped vertically. Defaults to false.
+  ///
+  /// If both are true, the child widget will be flipped both vertically and horizontally.
+  ///
+  /// The [alignment] controls the origin of the flip; by default, this is
+  /// the center of the box.
```

Comment:
What happens if the alignment is `Alignment.upperLeft` and both are flipped?  Does it just draw nothing?  Or does it draw outside of its bounding box?

Reply:
Actually, the more I think about it, isn't any alignment other than `center` going to draw nothing, or draw outside of its bounding box as long as one flip is true? That seems undesirable, since both are surprising.

Reply:
It draws outside the bounding box if alignment is something other than `center`. Being restrictive, we could just fix the alignment to `Alignment.center` and not provide the option to change it at all. But allowing it may help someone desiring the undesirable results. But they could also do it using the main `Transform()` constructor. But again, the purpose of this `flip` constructor is to just simplify and be descriptive of what is being done. What do you prefer?

Reply:
In that case, I think I'd prefer if we just eliminate the `alignment` argument altogether and just fix it at Alignment.center. It's surprising to have it draw outside of its bounding box, and if they want to do that, they can do it using the main constructor. We can always add it later if a compelling use case arises.